### PR TITLE
YOLOv5 & YOLOv8 support for the OpenVINO Detector

### PR DESCRIPTION
**Pull request by aeozyalcin**
I have added OpenVINO support for both YOLOv5 and YOLOv8 models. This diff adds on top of the previous work where YOLOX support was added. These are state of the art YOLO models, and I have observed incredible performance with them. Looking forward to seeing others try them out.

### CODE REVIEW

File: docs/docs/configuration/detectors.md [hunk 1]
```diff
@@ -118,8 +118,7 @@ model:
...
+This detector also supports some YOLO variants: YOLOX, YOLOv5, and YOLOv8 specifically. Other YOLO variants are not officially supported/tested. Frigate does not come with any yolo models preloaded, so you will need to supply your own models. This detector has been verified to work with the [yolox_tiny](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/yolox-tiny) model from Intel's Open Model Zoo. You can follow [these instructions](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/yolox-tiny#download-a-model-and-convert-it-into-openvino-ir-format) to retrieve the OpenVINO-compatible `yolox_tiny` model. Make sure that the model input dimensions match the `width` and `height` parameters, and `model_type` is set accordingly. See [Full Configuration Reference](index.md#full-configuration-reference) for a list of possible `model_type` options. Below is an example of how `yolox_tiny` can be used in Frigate:
```
Comment (NickM-27) [inline docs/docs/configuration/detectors.md:121]:
```suggestion

This detector also supports some YOLO variants: YOLOX, YOLOv5, and YOLOv8 specifically. Other YOLO variants are not officially supported/tested. Frigate does not come with any yolo models preloaded, so you will need to supply your own models. This detector has been verified to work with the [yolox_tiny](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/yolox-tiny) model from Intel's Open Model Zoo. You can follow [these instructions](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/yolox-tiny#download-a-model-and-convert-it-into-openvino-ir-format) to retrieve the OpenVINO-compatible `yolox_tiny` model. Make sure that the model input dimensions match the `width` and `height` parameters, and `model_type` is set accordingly. See [Full Configuration Reference](/configuration/index.md#full-configuration-reference) for a list of possible `model_type` options. Below is an example of how `yolox_tiny` can be used in Frigate:
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Review by Caros2017 — APPROVED** [review]

**Review by NickM-27 — APPROVED** [review]

**Comment by aeozyalcin** [comment]
Thanks y'all. I am considering adding some Google Collab notebook links in the docs for people to be able to retrieve and supply their own YOLO models to Frigate. I suspect there will be lots of questions about that, so a ready-made notebook would alleviate lots of those questions. Any thoughts?

File: frigate/detectors/plugins/openvino.py [hunk 2]
```diff
@@ -133,3 +133,68 @@ def detect_raw(self, tensor_input):
...
+            results = out_tensor.data[0]
+            output_data = np.transpose(results)
+            scores = np.max(output_data[:, 4:], axis=1)
+            if len(scores) == 0:
+                return np.zeros((20, 6), np.float32)
+            scores = np.expand_dims(scores, axis=1)
+            # add scores to the last column
+            dets = np.concatenate((output_data, scores), axis=1)
+            # filter out lines with scores below threshold
+            dets = dets[dets[:, -1] > 0.5, :]
```
Comment (NateMeyer) [inline frigate/detectors/plugins/openvino.py:147]: Any reason why this threshold is at 0.5?  This is different than the ssd and yolox models.

**Review by NateMeyer — COMMENTED** [review]
There's some duplicate code that could be refactored to make this more readable and maintainable.

File: frigate/detectors/plugins/openvino.py (hunk 2 above)
Comment (aeozyalcin) [reply frigate/detectors/plugins/openvino.py:147]: Performance reasons. YOLOv5 and v8 model output dimensions are usually on the order of thousands to tens of thousands arrays, depending on the model dimensions. So keeping the threshold a little higher showed significant performance improvements.

As an example, YOLOv8 inference+post processing takes ~33ms on my system with threshold of 0.5, while reducing the threshold to 0.3 increased the inference+post processing to ~45ms.

Ideally, we would limit to whatever `min_score` user has in their config, but each camera/area can have a different `min_score` setting, so I haven't thought of a good way to make use of that.

File: frigate/detectors/plugins/openvino.py [hunk 3]
```diff
@@ -133,3 +133,68 @@ def detect_raw(self, tensor_input):
...
+                        np.argmax(object_detected[4:-1]),  # Label ID
+                        object_detected[-1],  # Confidence
+                        (object_detected[1] - (object_detected[3] / 2))
+                        / self.h,  # y_min
+                        (object_detected[0] - (object_detected[2] / 2))
+                        / self.w,  # x_min
+                        (object_detected[1] + (object_detected[3] / 2))
+                        / self.h,  # y_max
+                        (object_detected[0] + (object_detected[2] / 2))
+                        / self.w,  # x_max
```
[the remark this answers is not in the document]
Comment (aeozyalcin) [reply frigate/detectors/plugins/openvino.py]: Take a look at the latest commit.

When rethinking and going over the existing code, I found that the SSD post-processing might have a bug, where if a detection with less than 0.1 score is encountered before all 20 spots in the detections array is filled up, the for loop will exit prematurely without going through the rest of the inference results, even though there may still be other inference results with confidence scores higher than 0.1. FYI.

**Review by NickM-27 — APPROVED** [review]

**Comment by aeozyalcin** [comment]
@NickM-27 the checks are stuck for some reason. Is there a way to kick them off again?

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**